tests/test_processing_stack.py

```diff
@@ -0,0 +1,126 @@
+import time
+
+from pytest import mark
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+# TODO: get resources from correct AWS stack (DEPLOY_ENV), if multiple stacks are deployed to
+# single AWS account
+ENV = os.environ.get("DEPLOY_ENV", "dev")
+
+
+@mark.infrastructure
+def test_should_return_healthy_compute_environment(batch_client):
+    datalake_compute_environment = {}
+
+    response = batch_client.describe_compute_environments()
+    for compute_environment in response["computeEnvironments"]:
+        if (
+            APPLICATION_TAG_KEY in compute_environment["tags"].keys()
+            and compute_environment["tags"][APPLICATION_TAG_KEY] == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Compute Environment: %s", datalake_compute_environment)
+            break
+
+    assert (
+        datalake_compute_environment["state"] == "ENABLED"
+        and datalake_compute_environment["status"] == "VALID"
+        and datalake_compute_environment["statusReason"] == "ComputeEnvironment Healthy"
+    )
+
+
+@mark.infrastructure
+def test_should_prove_compute_environment_is_working_by_running_single_job(batch_client):
+    datalake_job_definition = {}
+    datalake_job_queue = {}
+
+    job_definitions_response = batch_client.describe_job_definitions()
+    for job_definition in job_definitions_response["jobDefinitions"]:
+        if (
+            APPLICATION_TAG_KEY in job_definition["tags"].keys()
+            and job_definition["tags"][APPLICATION_TAG_KEY] == APPLICATION_TAG_VAL
+        ):
+            datalake_job_definition = job_definition
+            logger.info("Job Definition: %s", datalake_job_definition)
+            break
+
+    job_queues_response = batch_client.describe_job_queues()
+    for job_queue in job_queues_response["jobQueues"]:
+        if (
+            APPLICATION_TAG_KEY in job_queue["tags"].keys()
+            and job_queue["tags"][APPLICATION_TAG_KEY] == APPLICATION_TAG_VAL
+        ):
+            datalake_job_queue = job_queue
+            logger.info("Job Queue: %s", datalake_job_queue)
+            break
+
+    job_submit_response = batch_client.submit_job(
+        jobName="pytest",
+        jobDefinition=datalake_job_definition["jobDefinitionName"],
+        # jobDefinition="checkflatdirectorystruc-109bd0c41ea7464",
+        jobQueue=datalake_job_queue["jobQueueName"],
+    )
+    logger.info("Submitted Job: %s", job_submit_response)
+
+    job_status = None
+    while job_status not in ("FAILED", "SUCCEEDED"):
+        job_status_response = batch_client.describe_jobs(
+            jobs=[
+                job_submit_response["jobId"],
+            ]
+        )
+        logger.info("Running Job: %s", job_status_response)
+        job_status = job_status_response["jobs"][0]["status"]
+        time.sleep(5)
```

Comment:
We have [pytest-timeout](https://pypi.org/project/pytest-timeout/) installed; we could `mark.timeout(…)` to avoid this job running forever.

Alternatively we could pull things apart a bit and use a generic waiter. Some untested code:

```
def wait_for(predicate: Callable[..., bool], timeout: timedelta, interval: timedelta, *args: Any, **kwargs: Any) -> None:
    end = datetime.now() + timeout
    while datetime.now() < end:
        if predicate(*args, **kwargs):
            return
        sleep(interval.seconds)

    assert False, "Timeout"
```

Reply:
Done in e132cc941e96bbc1d50c24e3e208d6e2761090b3